mountpoint-s3-client/src/object_client.rs

```diff
@@ -245,9 +245,19 @@ pub enum GetObjectAttributesError {
 
 /// Parameters to a [ObjectClient::put_object] request
 /// TODO: Populate this struct with parameters from the S3 API, e.g., storage class, encryption.
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Clone)]
 #[non_exhaustive]
-pub struct PutObjectParams {}
+pub struct PutObjectParams {
+    pub trailing_checksums: bool,
+}
+
+impl Default for PutObjectParams {
+    fn default() -> Self {
+        Self {
+            trailing_checksums: true,
+        }
+    }
+}
```

Comment:
I don't think this should be on by default here -- as a generic S3 client it should be opt-in. But we should turn it on by default in `mountpoint_s3::Uploader`.